### PR TITLE
feat(twap): add recipient to advanced orders

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

- adds recipient to advanced-orders
- refactors settings code (move shared settings UI code to trade module so that it can be reused in limit and advanced orders)

# To test
- go to advanced orders
- open settings
- test recipient functionality
- do same for limit orders

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]
Great! Clear and structural!

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @nenadV91 , great job!
However, it would be great to enhance the order placement pop-up for a TWAP order when sending it to recipient:
to null
Recipient should be added to 'Expired', 'Cancelled', 'Traded' pop-ups as well. 

Then, it would be great to add a line 'Recipient' into the Order confirm modal (like we do on Limit orders page; and when a recipient differs from a sender's account)
recipient
 
Also, it would be great to clear out the field after an order placement 
image

File: src/modules/advancedOrders/containers/SettingsWidget/index.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { Settings } from 'modules/advancedOrders/pure/Settings'
+import { MenuContent, SettingsButton, SettingsIcon } from 'modules/trade/pure/Settings'
+
+import {
+  advancedOrdersSettingsAtom,
+  updateAdvancedOrdersSettingsAtom,
+  AdvancedOrdersSettingsState,
+} from '../../state/advancedOrdersSettingsAtom'
+
+export function AdvancedOrdersSettings() {
```
Comment (alfetopito) [inline src/modules/advancedOrders/containers/SettingsWidget/index.tsx:16]: Can you make the module name match the component?
Either make both SettingsWidget or AdvancedOrdersSettings

File: src/modules/advancedOrders/state/advancedOrdersSettingsAtom.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export interface AdvancedOrdersSettingsState {
+  readonly showRecipient: boolean
+}
+
+export const defaultAdvancedOrdersSettings: AdvancedOrdersSettingsState = {
+  showRecipient: false,
+}
+
+export const advancedOrdersSettingsAtom = atomWithStorage<AdvancedOrdersSettingsState>(
+  'advanced-orders-settings-atom:v2',
```
Comment (alfetopito) [inline src/modules/advancedOrders/state/advancedOrdersSettingsAtom.ts:13]: Can you start with `v0` instead of `v2`?

**Review by alfetopito — APPROVED** [review]
Very nice!
Just 2 small comments

Other than what Elena mentioned, nothing else from my side

File: src/modules/advancedOrders/containers/SettingsWidget/index.tsx (hunk 1 above)
Comment (nenadV91) [reply src/modules/advancedOrders/containers/SettingsWidget/index.tsx:16]: Makes sense, updated 👍

File: src/modules/advancedOrders/state/advancedOrdersSettingsAtom.ts (hunk 2 above)
Comment (nenadV91) [reply src/modules/advancedOrders/state/advancedOrdersSettingsAtom.ts:13]: Good catch!

**Review by alfetopito — APPROVED** [review]

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @nenadV91 , the order creation fails when I try creating an order to a recipient with ENS name :( 
image

File: src/modules/advancedOrders/updaters/RecipientUpdater.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { useUpdateAtom } from 'jotai/utils'
+import { useEffect } from 'react'
+
+import useENSAddress from 'legacy/hooks/useENSAddress'
+
+import { useAdvancedOrdersDerivedState } from '../hooks/useAdvancedOrdersDerivedState'
+import { updateAdvancedOrdersAtom } from '../state/advancedOrdersAtom'
+
+export function RecipientUpdater() {
```
Comment (shoom3301) [inline src/modules/advancedOrders/updaters/RecipientUpdater.tsx:9]: Great!

Comment (shoom3301) [reply src/modules/advancedOrders/updaters/RecipientUpdater.tsx:9]: Should we do the same for all trade widgets?

File: src/modules/twap/containers/TwapConfirmModal/TwapConfirmDetails.tsx [hunk 4]
```diff
@@ -39,6 +45,9 @@ export const TwapConfirmDetails = React.memo(function TwapConfirmDetails(props:
   const partDurationDisplay = partDuration ? deadlinePartsDisplay(partDuration, true) : ''
   const totalDurationDisplay = totalDuration ? deadlinePartsDisplay(totalDuration, true) : ''
 
+  const { account } = useAtomValue(walletInfoAtom)
```
Comment (shoom3301) [inline src/modules/twap/containers/TwapConfirmModal/TwapConfirmDetails.tsx:48]:
```suggestion
  const { account } = useWalletInfo()
```

File: src/modules/twap/containers/TwapConfirmModal/TwapConfirmDetails.tsx [hunk 5]
```diff
@@ -39,6 +45,9 @@ export const TwapConfirmDetails = React.memo(function TwapConfirmDetails(props:
...
+  const { recipient } = useAtomValue(advancedOrdersDerivedStateAtom)
```
Comment (shoom3301) [inline src/modules/twap/containers/TwapConfirmModal/TwapConfirmDetails.tsx:49]:
```suggestion
  const { recipient } = useAdvancedOrdersDerivedState()
```

**Review by shoom3301 — APPROVED** [review]

File: src/modules/advancedOrders/updaters/RecipientUpdater.tsx (hunk 3 above)
Comment (nenadV91) [reply src/modules/advancedOrders/updaters/RecipientUpdater.tsx]: Addressed this in https://github.com/cowprotocol/cowswap/pull/2886

**Pull request merged**